saleor/graphql/payment/mutations.py

```diff
@@ -55,8 +54,10 @@ class Meta:
 
     @classmethod
     def perform_mutation(cls, _root, info, checkout_id, **data):
-        checkout = cls.get_node_or_error(
-            info, checkout_id, field='checkout_id', only_type=Checkout)
+        checkout_id = cls.from_global_id_strict_type(
+            info, checkout_id, only_type=Checkout, field='checkout_id')
+        checkout = models.Checkout.objects.prefetch_related(
+            'lines__variant__product__collections').get(pk=checkout_id)
```

Comment:
Is there any way that we could use the optimizer here, maybe somehow with `Checkout.get_node`? I don't know if that's possible but it would be nice to be able to prefetch these fields only when they are requested in a query.

Reply:
I'm not sure of what you want there. It's a mutation, the lines collections are needed to get the discounted prices.

Reply:
OK, I thought this was added here to optimize the mutation response, not the actual operation. In this case, this is fine.